snap/component.go

```diff
@@ -41,17 +41,10 @@ type ComponentInfo struct {
 	// component is associated with. This field may be empty if the
 	// ComponentInfo was not created with the help of a snap.Info.
 	Hooks map[string]*HookInfo `yaml:"-"`
-}
 
-// NewComponentInfo creates a new ComponentInfo.
-func NewComponentInfo(cref naming.ComponentRef, ctype ComponentType, version, summary, description string) *ComponentInfo {
-	return &ComponentInfo{
-		Component:   cref,
-		Type:        ctype,
-		Version:     version,
-		Summary:     summary,
-		Description: description,
-	}
+	// ComponentSideInfo contains information for which the source of truth is
+	// not the component blob itself.
+	ComponentSideInfo
```

Comment:
Why do we need ComponentSideInfo now? I see that in the end you will have access to the revision from a SnapAppSet, but not sure if I see the use case in this PR, is that maybe in one of the follow-ups? Also, not sure why the revision would be needed from the interfaces code.